openapi-module/README.md

```diff
@@ -0,0 +1,152 @@
+
+# OpenAPI Module for Flexbench
+
+This module provides functionalities to parse OpenAPI documents and generate test scenarios for Flexbench.
+
+## Prerequisites
+
+- To use this module, you need to install Node.js. It is recommended to use the latest LTS version available.
+  Please install Node.js using pre-built installers for your platform to avoid incompatibility issues with different development tools.
+
+### To check that Node.js was installed correctly, type the following commands in your terminal client:
+
+```sh
+node -v
+npm -v
+```
+
+## Project Structure
+
+```
+openapi-module/
+├── node_modules/
+├── sample/
+│   └── sample-openapi.yaml
+├── scripts/
+│   ├── generate-all.js
+│   ├── generate-curl.js
+│   └── generate-flex.js
+├── src/
+│   ├── generators/
+│   │   ├── curl-generator.js
+│   │   ├── field-mapping.js
+│   │   ├── flex-generator.js
+│   │   └── gpt-flex-generator.js
+│   ├── GPT/
+│   │   └── config.js
+│   └── parsers/
+│       └── openapi-parser.js
+├── temp/(generated files would be save under this folder)
+├── test/
+│   ├── generators.test.js
+│   └── parser.test.js
+├── .gitignore
+├── package-lock.json
+├── package.json
+└── README.md
+```
+
+## Installation
+
+### To install dependencies, run the following command:
+
+```sh
+npm install
+```
+
+## Usage
+
+### Before generate .flex file & curl-cmd for Flexbench by parsing your OpenAPI yaml:
+
+Please decide to generate scenarios by AI or static approach:
```

Comment:
I have added my own openAI key, but I could not generate the flex gpt file. I was either not getting an error message or getting a connection error:

```
iskitsas001@GR_D36WYRX1MV openapi-module % set -a
iskitsas001@GR_D36WYRX1MV openapi-module % export OPENAI_API_KEY='...FaWLw-xoC...'
iskitsas001@GR_D36WYRX1MV openapi-module % npm run generate-flex -- --openApiFilePath=sample/sample-openapi.yaml --outputFilePath=temp/flex-scenarios3.json

> flexbench-openapi-module@1.0.0 generate-flex
> node scripts/generate-flex.js --openApiFilePath=sample/sample-openapi.yaml --outputFilePath=temp/flex-scenarios3.json

OpenAI API error: undefined Connection error. undefined undefined
Failed to generate Flex scenarios using GPT.
Flex scenario generation with GPT complete.

```

When was not getting a connection error, the file was not generated

Reply:
it runs successfully at my end:
```
aj@AJs-Mac openapi-module % npm run generate-flex -- --openApiFilePath=sample/sample-openapi.yaml --outputFilePath=temp/flex-scenarios.json --useGPT=true --gptOutputFilename=my-custom-scenario.flex


> flexbench-openapi-module@1.0.0 generate-flex
> node scripts/generate-flex.js --openApiFilePath=sample/sample-openapi.yaml --outputFilePath=temp/flex-scenarios.json --useGPT=true --gptOutputFilename=my-custom-scenario.flex

Directory created or already exists: /Users/aj/GSOC/flexbench/openapi-module/temp
Flex scenarios generated and saved to /Users/aj/GSOC/flexbench/openapi-module/temp/my-custom-scenario.flex
Flex scenario generation with GPT complete.
```
generated: 
```
{
  "project": {
    "projectName": "Flexbench Project",
    "description": "API Load Testing Project",
    "_id": "fde5e6ea-5ce4-4b6f-bd9c-1c7e3a7e01a6"
  },
  "scenarios": [
    {
      "projectId": "fde5e6ea-5ce4-4b6f-bd9c-1c7e3a7e01a6",
      "scenarioname": "User Scenario",
      "duration": "5",
      "workers": "4",
      "totalclients": "10",
      "throttling": "50000",
      "delay": "1.0",
      "_id": "cde8b2a7-8d0c-4c9e-ae8a-8b1d7c7fa546"
    },
    {
      "projectId": "fde5e6ea-5ce4-4b6f-bd9c-1c7e3a7e01a6",
      "scenarioname": "User Creation Scenario",
      "duration": "5",
      "workers": "4",
      "totalclients": "10",
      "throttling": "50000",
      "delay": "1.0",
      "_id": "a1f8e3c9-7b6a-4e9d-aa5c-2d1f9d1e3c8b"
    }
  ],
  "requests": [
    {
      "scenarioId": "cde8b2a7-8d0c-4c9e-ae8a-8b1d7c7fa546",
      "requestName": "Get Users",
      "url": "localhost:4000",
      "protocol": "",
      "host": "localhost",
      "method": "GET",
      "path": "/users",
      "port": "4000",
      "body": [],
      "header": [
        {
          "key": "Content-Type",
          "value": "application/json",
          "description": ""
        }
      ],
      "_id": "e3bb0f56-35e5-4f4f-af67-cc2c8cb20ab9"
    },
    {
      "scenarioId": "a1f8e3c9-7b6a-4e9d-aa5c-2d1f9d1e3c8b",
      "requestName": "Create User",
      "url": "localhost:4000",
      "protocol": "",
      "host": "localhost",
      "method": "POST",
      "path": "/users",
      "port": "4000",
      "body": [
        {
          "key": "name",
          "value": "John Doe",
          "type": "TEXT",
          "description": ""
        },
        {
          "key": "email",
          "value": "johndoe@example.com",
          "type": "TEXT",
          "description": ""
        }
      ],
      "header": [
        {
          "key": "Content-Type",
          "value": "application/json",
          "description": ""
        }
      ],
      "_id": "1d2f3a4b-5c6d-7e8f-9a0b-2c3d4e5f6a7b"
    }
  ]
}
```

Reply:
connection issue may caused by the fact that OpenAI requires a payment method set up before you use your API key.